pdp/static/js/pdp_raster_map.js

```diff
@@ -139,15 +178,42 @@ function processNcwmsLayerMetadata(ncwms_layer, catalog) {
 function setTimeAvailable(begin, end) {
     //TODO: only present times available in ncwms capabilities for this layer
     var yearRange = begin.getFullYear().toString(10) + ":" + end.getFullYear().toString(10);
-
+        
+    //preserve an active range previously set by a user to faciliate downloading matched data.
+    var previousMinimum = $(".datepickerstart").datepicker("option", "minDate");
+    var previousMaximum = $(".datepickerend").datepicker("option", "maxDate");
+    var previousRangeFrom = $(".datepickerstart").datepicker("getDate");
+    var previousRangeTo = $(".datepickerend").datepicker("getDate");
+    
+    //set new maximums and minimums
     $.each([".datepickerstart", ".datepickerend"], function (idx, val) {
         $(val).datepicker("option", "minDate", begin);
         $(val).datepicker("option", "maxDate", end);
         $(val).datepicker("option", "yearRange", yearRange);
     });
-    $(".datepicker").datepicker("setDate", begin);
-    $(".datepickerstart").datepicker("setDate", begin);
-    $(".datepickerend").datepicker("setDate", end);
+
+    //try to keep the active range, if it was specified and is possible.
+    //fall back to the beginning and end of the new dataset.
+    if(previousMinimum 
```

Comment:
`previousMinimum` is `null` during the initial page load, the first time through this function.